src/Plugin/CKEditorPlugin/DrupalEntity.php

```diff
@@ -42,13 +44,26 @@ class DrupalEntity extends CKEditorPluginBase {
    *   The plugin_id for the plugin instance.
    * @param mixed $plugin_definition
    *   The plugin implementation definition.
+   * @param \Drupal\Core\Entity\Query\QueryFactory $entity_query
+   *   The entity query.
    */
-  public function __construct(array $configuration, $plugin_id, $plugin_definition) {
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, QueryFactory $entity_query) {
     parent::__construct($configuration, $plugin_id, $plugin_definition);
 
-    $this->embedButtons = \Drupal::entityQuery('embed_button')->execute();
+    $this->embedButtons = $entity_query->get('embed_button')->execute();
```

Comment:
This currently doesn't work. The class will need to implement `ContainerFactoryPluginInterface` in order for the create function to be executed. Also consider passing just embed_button entity query object instead of the entire entity.query service